Sources/MapboxCoreNavigation/RouteLegProgress.swift

```diff
@@ -144,8 +144,6 @@ open class RouteLegProgress: Codable {
         self.leg = leg
         self.stepIndex = stepIndex
         
-        precondition(leg.steps.indices.contains(stepIndex), "It's not possible to create RouteLegProgress without any steps or when stepIndex is higher than steps count.")
```

Comment:
Why was this precondition removed? `didSet` for `stepIndex` won't be triggered in init.

Reply:
It's removed because it's after we give value for the `stepIndex `. I'll move it back here.